src/array/specification.rs

```diff
@@ -75,28 +75,47 @@ pub fn check_offsets_minimal<O: Offset>(offsets: &[O], values_len: usize) -> usi
 /// * any slice of `values` between two consecutive pairs from `offsets` is invalid `utf8`, or
 /// * any offset is larger or equal to `values_len`.
 pub fn check_offsets_and_utf8<O: Offset>(offsets: &[O], values: &[u8]) {
-    offsets.windows(2).for_each(|window| {
-        let start = window[0].to_usize();
-        let end = window[1].to_usize();
-        // assert monotonicity
-        assert!(start <= end);
-        // assert bounds
-        let slice = &values[start..end];
-        // assert utf8
-        simdutf8::basic::from_utf8(slice).expect("A non-utf8 string was passed.");
-    });
+    const SIMD_CHUNK_SIZE: usize = 64;
+
+    let all_ascii = values.is_ascii();
```

Comment:
FYI @ritchie46 this implementation should be faster than writing a manual loop with `all` for larger buffers.